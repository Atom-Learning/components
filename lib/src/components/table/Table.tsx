import * as React from 'react'

import { styled } from '~/stitches'

import { TableBody } from './TableBody'
import { TableCell } from './TableCell'
import { TableFooter } from './TableFooter'
import { TableFooterCell } from './TableFooterCell'
import { TableHeader } from './TableHeader'
import { TableHeaderCell } from './TableHeaderCell'
import { StyledRow, TableRow } from './TableRow'
import { TableStickyColumnsContainer } from './TableStickyColumnsContainer'

type TableSubComponents = {
  Body: typeof TableBody
  Cell: typeof TableCell
  Footer: typeof TableFooter
  FooterCell: typeof TableFooterCell
  Header: typeof TableHeader
  HeaderCell: typeof TableHeaderCell
  Row: typeof TableRow
  StickyColumnsContainer: typeof TableStickyColumnsContainer
}

const StyledTable = styled('table', {
  borderCollapse: 'separate',
  borderSpacing: 0,
  fontFamily: '$sans',
  fontSize: '$sm',
  width: '100%',
  variants: {
    size: {
      md: {
        [`${TableCell}, ${TableHeaderCell}, ${TableFooterCell}`]: {
          height: '$4'
        }
      },
      lg: {
        [`${TableCell}, ${TableHeaderCell}, ${TableFooterCell}`]: {
          height: '$5'
        }
      }
    },
    corners: {
      round: {
        [`${TableHeaderCell}`]: {
          '&:first-of-type': { borderTopLeftRadius: '$0' },
          '&:last-of-type': { borderTopRightRadius: '$0' }
        },
        [`${StyledRow}:last-child`]: {
          [`${TableCell}:first-child`]: { borderBottomLeftRadius: '$0' },
          [`${TableCell}:last-child`]: { borderBottomRightRadius: '$0' }
        }
      },
      square: {}
    }
  }
})

type TableProps = React.ComponentProps<typeof StyledTable>

export const Table: React.FC<TableProps> & TableSubComponents = ({
  size = 'md',
  corners = 'round',
  ...rest
}: TableProps) => <StyledTable size={size} corners={corners} {...rest} />

Table.Body = TableBody
Table.Cell = TableCell
Table.Footer = TableFooter
Table.FooterCell = TableFooterCell
Table.Header = TableHeader
Table.HeaderCell = TableHeaderCell
Table.Row = TableRow
Table.StickyColumnsContainer = TableStickyColumnsContainer

Table.displayName = 'Table'
