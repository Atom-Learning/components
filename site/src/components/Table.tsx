import { styled } from '@atom-learning/components'

export const Table = styled('table', {
  borderCollapse: 'collapse',
  width: '100%'
})
export const Cell = styled('td', {
  borderBottom: '1px solid $tonal200',
  fontFamily: '$sans',
  pr: '$4',
  py: '$4',
  textAlign: 'left',
  verticalAlign: 'middle',
  variants: {
    appearance: {
      heading: {
        color: '$tonal700',
        fontWeight: 700,
        whiteSpace: 'nowrap'
      },
      content: {
        color: '$tonal600'
      }
    },
    size: {
      sm: {
        fontSize: '$sm'
      },
      md: {
        fontSize: '$md'
      }
    }
  },
  defaultVariants: {
    appearance: 'content',
    size: 'sm'
  }
})
