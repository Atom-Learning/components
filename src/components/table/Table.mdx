---
title: Table
component: Table, Table.Body, Table.Cell, Table.Footer, Table.Header, Table.HeaderCell, Table.Row
description: The Table component displays a collection of data grouped into rows
category: Layout
---

The `Table` component structure mirrors that of a regular HTML table, with the smaller `Table.Body`, `Table.Cell`, `Table.Footer`, `Table.Header`, `Table.HeaderCell` and `Table Row` components corresponding to the `<tbody>`, `<td>`, `<tfoot>`, `<thead>`, `<th>` and `<tr>` tags, respectively. 

The default table width is 100% of the parent container, but this and other styles can be overridden with the `css` prop on all components.

Best practice: Due to rendering issues with some browsers, adding borders to `Table.Row` is possible by styling the Table component with `border-collapse: collapse`; however, this practice is generally not recommended. The preferred method for adding borders is applying styles to the `Table.Cell` and `Table.HeaderCell` components.


```jsx preview
<Table css={{ height: 100, width: 400, color: '$primary500' }}>
  <Table.Header>
    <Table.Row>
      <Table.HeaderCell>First Name</Table.HeaderCell>
      <Table.HeaderCell>Last Name</Table.HeaderCell>
      <Table.HeaderCell>Age</Table.HeaderCell>
    </Table.Row>
  </Table.Header>
  <Table.Body>
    <Table.Row>
      <Table.Cell css={{border: '1px solid $tonal300'}}>Rakim</Table.Cell>
      <Table.Cell css={{border: '1px solid $tonal300'}}>Jackson</Table.Cell>
      <Table.Cell css={{border: '1px solid $tonal300'}}>35</Table.Cell>
    </Table.Row>
    <Table.Row>
      <Table.Cell css={{border: '1px solid $tonal300'}}>Evelyn</Table.Cell>
      <Table.Cell css={{border: '1px solid $tonal300'}}>Smith</Table.Cell>
      <Table.Cell css={{border: '1px solid $tonal300'}}>27</Table.Cell>
    </Table.Row>
  </Table.Body>
  <Table.Footer>
    <Table.Row>
      <Table.Cell>Footer 1</Table.Cell>
    </Table.Row>
  </Table.Footer>
</Table>
```
