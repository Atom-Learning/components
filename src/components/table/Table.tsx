import * as React from 'react'

import { styled } from '~/stitches'

import { TableBody } from './TableBody'
import { TableCell } from './TableCell'
import { TableFooter } from './TableFooter'
import { TableFooterCell } from './TableFooterCell'
import { TableHeader } from './TableHeader'
import { TableHeaderCell } from './TableHeaderCell'
import { TableRow } from './TableRow'

type TableSubComponents = {
  Body: typeof TableBody
  Cell: typeof TableCell
  Footer: typeof TableFooter
  FooterCell: typeof TableFooterCell
  Header: typeof TableHeader
  HeaderCell: typeof TableHeaderCell
  Row: typeof TableRow
}

const StyledTable = styled('table', {
  borderCollapse: 'collapse',
  fontFamily: '$sans',
  fontSize: '$sm',
  width: '100%',
  variants: {
    size: {
      md: {
        [`${TableCell}`]: {
          height: '$4'
        },
        [`${TableHeaderCell}`]: {
          height: '$4'
        }
      },
      lg: {
        [`${TableCell}`]: {
          height: '$5'
        },
        [`${TableHeaderCell}`]: {
          height: '$5'
        }
      }
    }
  }
})

type TableProps = React.ComponentProps<typeof StyledTable>

export const Table: React.FC<TableProps> & TableSubComponents = ({
  size = 'md',
  ...rest
}: TableProps) => <StyledTable size={size} {...rest} />

Table.Body = TableBody
Table.Cell = TableCell
Table.Footer = TableFooter
Table.FooterCell = TableFooterCell
Table.Header = TableHeader
Table.HeaderCell = TableHeaderCell
Table.Row = TableRow

Table.displayName = 'Table'
