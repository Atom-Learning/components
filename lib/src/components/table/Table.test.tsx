import { render } from '@testing-library/react'
import { axe } from 'jest-axe'
import * as React from 'react'

import { Table } from './Table'

describe(`Table component`, () => {
  it('renders', async () => {
    const { container } = await render(
      <Table css={{ height: '100px', width: '400px' }}>
        <Table.Header>
          <Table.Row>
            <Table.HeaderCell>Column A</Table.HeaderCell>
          </Table.Row>
        </Table.Header>
        <Table.Body>
          <Table.Row>
            <Table.Cell>This is text</Table.Cell>
          </Table.Row>
        </Table.Body>
        <Table.Footer>
          <Table.Row>
            <Table.Cell>Footer 1</Table.Cell>
          </Table.Row>
        </Table.Footer>
      </Table>
    )
    expect(container).toMatchSnapshot()
  })

  it('renders with size set to lg', async () => {
    const { container } = await render(
      <Table size="lg" css={{ height: '100px', width: '400px' }}>
        <Table.Header>
          <Table.Row>
            <Table.HeaderCell>Column A</Table.HeaderCell>
          </Table.Row>
        </Table.Header>
        <Table.Body>
          <Table.Row>
            <Table.Cell>This is text</Table.Cell>
          </Table.Row>
        </Table.Body>
        <Table.Footer>
          <Table.Row>
            <Table.Cell>Footer 1</Table.Cell>
          </Table.Row>
        </Table.Footer>
      </Table>
    )
    expect(container).toMatchSnapshot()
  })

  it('renders with size set to xl', async () => {
    const { container } = await render(
      <Table size="xl" css={{ height: '100px', width: '400px' }}>
        <Table.Header>
          <Table.Row>
            <Table.HeaderCell>Column A</Table.HeaderCell>
          </Table.Row>
        </Table.Header>
        <Table.Body>
          <Table.Row>
            <Table.Cell>This is text</Table.Cell>
          </Table.Row>
        </Table.Body>
        <Table.Footer>
          <Table.Row>
            <Table.Cell>Footer 1</Table.Cell>
          </Table.Row>
        </Table.Footer>
      </Table>
    )
    expect(container).toMatchSnapshot()
  })

  it('renders with square corners', async () => {
    const { container } = await render(
      <Table corners="square" css={{ height: '100px', width: '400px' }}>
        <Table.Header>
          <Table.Row>
            <Table.HeaderCell>Column A</Table.HeaderCell>
          </Table.Row>
        </Table.Header>
        <Table.Body>
          <Table.Row>
            <Table.Cell>This is text</Table.Cell>
          </Table.Row>
        </Table.Body>
        <Table.Footer>
          <Table.Row>
            <Table.Cell>Footer 1</Table.Cell>
          </Table.Row>
        </Table.Footer>
      </Table>
    )
    expect(container).toMatchSnapshot()
  })

  it('renders with a themed header', async () => {
    const { container } = await render(
      <Table css={{ height: '100px', width: '400px' }}>
        <Table.Header theme="primary">
          <Table.Row>
            <Table.HeaderCell>Column A</Table.HeaderCell>
          </Table.Row>
        </Table.Header>
        <Table.Body>
          <Table.Row>
            <Table.Cell>This is text</Table.Cell>
          </Table.Row>
        </Table.Body>
        <Table.Footer>
          <Table.Row>
            <Table.Cell>Footer 1</Table.Cell>
          </Table.Row>
        </Table.Footer>
      </Table>
    )
    expect(container).toMatchSnapshot()
  })

  it('has no programmatically detectable a11y issues', async () => {
    const { container } = render(
      <Table>
        <Table.Header>
          <Table.Row>
            <Table.HeaderCell>Column A</Table.HeaderCell>
          </Table.Row>
        </Table.Header>
        <Table.Body>
          <Table.Row>
            <Table.Cell>This is text</Table.Cell>
          </Table.Row>
        </Table.Body>
        <Table.Footer>
          <Table.Row>
            <Table.Cell>Footer 1</Table.Cell>
          </Table.Row>
        </Table.Footer>
      </Table>
    )
    expect(await axe(container)).toHaveNoViolations()
  })

  it('renders with sticky columns', async () => {
    const numberOfStickyColumns = 1
    const { container } = render(
      <Table
        css={{ height: '100px', width: '300px' }}
        numberOfStickyColumns={numberOfStickyColumns}
      >
        <Table.Header numberOfStickyColumns={numberOfStickyColumns}>
          <Table.Row>
            <Table.HeaderCell>Column A</Table.HeaderCell>
            <Table.HeaderCell>Column B</Table.HeaderCell>
            <Table.HeaderCell>Column C</Table.HeaderCell>
          </Table.Row>
        </Table.Header>
        <Table.Body numberOfStickyColumns={numberOfStickyColumns}>
          <Table.Row>
            <Table.Cell>This is text</Table.Cell>
            <Table.Cell>This is text</Table.Cell>
            <Table.Cell>This is text</Table.Cell>
          </Table.Row>
        </Table.Body>
      </Table>
    )
    expect(container).toMatchSnapshot()
  })
})
