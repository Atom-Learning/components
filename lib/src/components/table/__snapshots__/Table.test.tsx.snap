// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table component renders 1`] = `
@media  {
  .c-cwQMhQ {
    border-collapse: separate;
    border-spacing: 0;
    font-family: var(--fonts-sans);
    font-size: var(--fontSizes-sm);
    width: 100%;
  }

  .c-hjAYDb {
    background: unset;
  }

  .c-GSaiK {
    color: white;
    font-family: var(--fonts-body);
    font-weight: 600;
    line-height: 1.5;
    padding: var(--space-2) var(--space-3);
    text-align: left;
    vertical-align: middle;
  }

  .c-cCZzXk {
    border-bottom: 1px solid var(--colors-tonal100);
    box-sizing: border-box;
    color: var(--colors-tonal400);
    font-family: var(--fonts-body);
    line-height: 1.5;
    padding: var(--space-2) var(--space-3);
    text-align: left;
    vertical-align: middle;
  }

  .c-cCZzXk:first-child {
    font-weight: bold;
  }
}

@media  {
  .c-cwQMhQ-gWDrUk-size-md .c-cCZzXk,
  .c-cwQMhQ-gWDrUk-size-md .c-GSaiK,
  .c-cwQMhQ-gWDrUk-size-md .c-hYjVQ {
    height: var(--sizes-4);
  }

  .c-cwQMhQ-codIxD-corners-round .c-GSaiK:first-of-type {
    border-top-left-radius: var(--radii-0);
  }

  .c-cwQMhQ-codIxD-corners-round .c-GSaiK:last-of-type {
    border-top-right-radius: var(--radii-0);
  }

  .c-cwQMhQ-codIxD-corners-round [object Object]:last-child .c-cCZzXk:first-child {
    border-bottom-left-radius: var(--radii-0);
  }

  .c-cwQMhQ-codIxD-corners-round [object Object]:last-child .c-cCZzXk:last-child {
    border-bottom-right-radius: var(--radii-0);
  }

  .c-PJLV-jEDwTF-theme-primaryDark .c-GSaiK {
    background: var(--colors-primaryDark);
  }

  .c-PJLV-gbDJOS-striped-true [object Object]:nth-child(odd) {
    background: white;
  }

  .c-PJLV-gbDJOS-striped-true [object Object]:nth-child(even) {
    background: var(--colors-tonal50);
  }
}

@media  {
  .c-cwQMhQ-icnGIKd-css {
    height: 100px;
    width: 400px;
  }
}

<div>
  <table
    class="c-cwQMhQ c-cwQMhQ-gWDrUk-size-md c-cwQMhQ-codIxD-corners-round c-cwQMhQ-icnGIKd-css"
  >
    <thead
      class="c-PJLV c-PJLV-jEDwTF-theme-primaryDark"
    >
      <tr
        class="c-hjAYDb"
      >
        <th
          class="c-GSaiK"
        >
          Column A
        </th>
      </tr>
    </thead>
    <tbody
      class="c-PJLV c-PJLV-gbDJOS-striped-true"
    >
      <tr
        class="c-hjAYDb"
      >
        <td
          class="c-cCZzXk"
        >
          This is text
        </td>
      </tr>
    </tbody>
    <tfoot
      class="c-PJLV"
    >
      <tr
        class="c-hjAYDb"
      >
        <td
          class="c-cCZzXk"
        >
          Footer 1
        </td>
      </tr>
    </tfoot>
  </table>
</div>
`;

exports[`Table component renders with a themed header 1`] = `
@media  {
  .c-cwQMhQ {
    border-collapse: separate;
    border-spacing: 0;
    font-family: var(--fonts-sans);
    font-size: var(--fontSizes-sm);
    width: 100%;
  }

  .c-hjAYDb {
    background: unset;
  }

  .c-GSaiK {
    color: white;
    font-family: var(--fonts-body);
    font-weight: 600;
    line-height: 1.5;
    padding: var(--space-2) var(--space-3);
    text-align: left;
    vertical-align: middle;
  }

  .c-cCZzXk {
    border-bottom: 1px solid var(--colors-tonal100);
    box-sizing: border-box;
    color: var(--colors-tonal400);
    font-family: var(--fonts-body);
    line-height: 1.5;
    padding: var(--space-2) var(--space-3);
    text-align: left;
    vertical-align: middle;
  }

  .c-cCZzXk:first-child {
    font-weight: bold;
  }
}

@media  {
  .c-cwQMhQ-gWDrUk-size-md .c-cCZzXk,
  .c-cwQMhQ-gWDrUk-size-md .c-GSaiK,
  .c-cwQMhQ-gWDrUk-size-md .c-hYjVQ {
    height: var(--sizes-4);
  }

  .c-cwQMhQ-codIxD-corners-round .c-GSaiK:first-of-type {
    border-top-left-radius: var(--radii-0);
  }

  .c-cwQMhQ-codIxD-corners-round .c-GSaiK:last-of-type {
    border-top-right-radius: var(--radii-0);
  }

  .c-cwQMhQ-codIxD-corners-round [object Object]:last-child .c-cCZzXk:first-child {
    border-bottom-left-radius: var(--radii-0);
  }

  .c-cwQMhQ-codIxD-corners-round [object Object]:last-child .c-cCZzXk:last-child {
    border-bottom-right-radius: var(--radii-0);
  }

  .c-PJLV-jEDwTF-theme-primaryDark .c-GSaiK {
    background: var(--colors-primaryDark);
  }

  .c-PJLV-gbDJOS-striped-true [object Object]:nth-child(odd) {
    background: white;
  }

  .c-PJLV-gbDJOS-striped-true [object Object]:nth-child(even) {
    background: var(--colors-tonal50);
  }

  .c-cwQMhQ-goJwcl-size-lg .c-cCZzXk,
  .c-cwQMhQ-goJwcl-size-lg .c-GSaiK,
  .c-cwQMhQ-goJwcl-size-lg .c-hYjVQ {
    height: var(--sizes-5);
  }

  .c-PJLV-gmYsCp-theme-primary .c-GSaiK {
    background: var(--colors-primary);
  }
}

@media  {
  .c-cwQMhQ-icnGIKd-css {
    height: 100px;
    width: 400px;
  }
}

<div>
  <table
    class="c-cwQMhQ c-cwQMhQ-gWDrUk-size-md c-cwQMhQ-codIxD-corners-round c-cwQMhQ-icnGIKd-css"
  >
    <thead
      class="c-PJLV c-PJLV-gmYsCp-theme-primary"
    >
      <tr
        class="c-hjAYDb"
      >
        <th
          class="c-GSaiK"
        >
          Column A
        </th>
      </tr>
    </thead>
    <tbody
      class="c-PJLV c-PJLV-gbDJOS-striped-true"
    >
      <tr
        class="c-hjAYDb"
      >
        <td
          class="c-cCZzXk"
        >
          This is text
        </td>
      </tr>
    </tbody>
    <tfoot
      class="c-PJLV"
    >
      <tr
        class="c-hjAYDb"
      >
        <td
          class="c-cCZzXk"
        >
          Footer 1
        </td>
      </tr>
    </tfoot>
  </table>
</div>
`;

exports[`Table component renders with size set to lg 1`] = `
@media  {
  .c-cwQMhQ {
    border-collapse: separate;
    border-spacing: 0;
    font-family: var(--fonts-sans);
    font-size: var(--fontSizes-sm);
    width: 100%;
  }

  .c-hjAYDb {
    background: unset;
  }

  .c-GSaiK {
    color: white;
    font-family: var(--fonts-body);
    font-weight: 600;
    line-height: 1.5;
    padding: var(--space-2) var(--space-3);
    text-align: left;
    vertical-align: middle;
  }

  .c-cCZzXk {
    border-bottom: 1px solid var(--colors-tonal100);
    box-sizing: border-box;
    color: var(--colors-tonal400);
    font-family: var(--fonts-body);
    line-height: 1.5;
    padding: var(--space-2) var(--space-3);
    text-align: left;
    vertical-align: middle;
  }

  .c-cCZzXk:first-child {
    font-weight: bold;
  }
}

@media  {
  .c-cwQMhQ-gWDrUk-size-md .c-cCZzXk,
  .c-cwQMhQ-gWDrUk-size-md .c-GSaiK,
  .c-cwQMhQ-gWDrUk-size-md .c-hYjVQ {
    height: var(--sizes-4);
  }

  .c-cwQMhQ-codIxD-corners-round .c-GSaiK:first-of-type {
    border-top-left-radius: var(--radii-0);
  }

  .c-cwQMhQ-codIxD-corners-round .c-GSaiK:last-of-type {
    border-top-right-radius: var(--radii-0);
  }

  .c-cwQMhQ-codIxD-corners-round [object Object]:last-child .c-cCZzXk:first-child {
    border-bottom-left-radius: var(--radii-0);
  }

  .c-cwQMhQ-codIxD-corners-round [object Object]:last-child .c-cCZzXk:last-child {
    border-bottom-right-radius: var(--radii-0);
  }

  .c-PJLV-jEDwTF-theme-primaryDark .c-GSaiK {
    background: var(--colors-primaryDark);
  }

  .c-PJLV-gbDJOS-striped-true [object Object]:nth-child(odd) {
    background: white;
  }

  .c-PJLV-gbDJOS-striped-true [object Object]:nth-child(even) {
    background: var(--colors-tonal50);
  }

  .c-cwQMhQ-goJwcl-size-lg .c-cCZzXk,
  .c-cwQMhQ-goJwcl-size-lg .c-GSaiK,
  .c-cwQMhQ-goJwcl-size-lg .c-hYjVQ {
    height: var(--sizes-5);
  }
}

@media  {
  .c-cwQMhQ-icnGIKd-css {
    height: 100px;
    width: 400px;
  }
}

<div>
  <table
    class="c-cwQMhQ c-cwQMhQ-goJwcl-size-lg c-cwQMhQ-codIxD-corners-round c-cwQMhQ-icnGIKd-css"
  >
    <thead
      class="c-PJLV c-PJLV-jEDwTF-theme-primaryDark"
    >
      <tr
        class="c-hjAYDb"
      >
        <th
          class="c-GSaiK"
        >
          Column A
        </th>
      </tr>
    </thead>
    <tbody
      class="c-PJLV c-PJLV-gbDJOS-striped-true"
    >
      <tr
        class="c-hjAYDb"
      >
        <td
          class="c-cCZzXk"
        >
          This is text
        </td>
      </tr>
    </tbody>
    <tfoot
      class="c-PJLV"
    >
      <tr
        class="c-hjAYDb"
      >
        <td
          class="c-cCZzXk"
        >
          Footer 1
        </td>
      </tr>
    </tfoot>
  </table>
</div>
`;

exports[`Table component renders with square corners 1`] = `
@media  {
  .c-cwQMhQ {
    border-collapse: separate;
    border-spacing: 0;
    font-family: var(--fonts-sans);
    font-size: var(--fontSizes-sm);
    width: 100%;
  }

  .c-hjAYDb {
    background: unset;
  }

  .c-GSaiK {
    color: white;
    font-family: var(--fonts-body);
    font-weight: 600;
    line-height: 1.5;
    padding: var(--space-2) var(--space-3);
    text-align: left;
    vertical-align: middle;
  }

  .c-cCZzXk {
    border-bottom: 1px solid var(--colors-tonal100);
    box-sizing: border-box;
    color: var(--colors-tonal400);
    font-family: var(--fonts-body);
    line-height: 1.5;
    padding: var(--space-2) var(--space-3);
    text-align: left;
    vertical-align: middle;
  }

  .c-cCZzXk:first-child {
    font-weight: bold;
  }
}

@media  {
  .c-cwQMhQ-gWDrUk-size-md .c-cCZzXk,
  .c-cwQMhQ-gWDrUk-size-md .c-GSaiK,
  .c-cwQMhQ-gWDrUk-size-md .c-hYjVQ {
    height: var(--sizes-4);
  }

  .c-cwQMhQ-codIxD-corners-round .c-GSaiK:first-of-type {
    border-top-left-radius: var(--radii-0);
  }

  .c-cwQMhQ-codIxD-corners-round .c-GSaiK:last-of-type {
    border-top-right-radius: var(--radii-0);
  }

  .c-cwQMhQ-codIxD-corners-round [object Object]:last-child .c-cCZzXk:first-child {
    border-bottom-left-radius: var(--radii-0);
  }

  .c-cwQMhQ-codIxD-corners-round [object Object]:last-child .c-cCZzXk:last-child {
    border-bottom-right-radius: var(--radii-0);
  }

  .c-PJLV-jEDwTF-theme-primaryDark .c-GSaiK {
    background: var(--colors-primaryDark);
  }

  .c-PJLV-gbDJOS-striped-true [object Object]:nth-child(odd) {
    background: white;
  }

  .c-PJLV-gbDJOS-striped-true [object Object]:nth-child(even) {
    background: var(--colors-tonal50);
  }

  .c-cwQMhQ-goJwcl-size-lg .c-cCZzXk,
  .c-cwQMhQ-goJwcl-size-lg .c-GSaiK,
  .c-cwQMhQ-goJwcl-size-lg .c-hYjVQ {
    height: var(--sizes-5);
  }
}

@media  {
  .c-cwQMhQ-icnGIKd-css {
    height: 100px;
    width: 400px;
  }
}

<div>
  <table
    class="c-cwQMhQ c-cwQMhQ-gWDrUk-size-md c-cwQMhQ-icnGIKd-css"
  >
    <thead
      class="c-PJLV c-PJLV-jEDwTF-theme-primaryDark"
    >
      <tr
        class="c-hjAYDb"
      >
        <th
          class="c-GSaiK"
        >
          Column A
        </th>
      </tr>
    </thead>
    <tbody
      class="c-PJLV c-PJLV-gbDJOS-striped-true"
    >
      <tr
        class="c-hjAYDb"
      >
        <td
          class="c-cCZzXk"
        >
          This is text
        </td>
      </tr>
    </tbody>
    <tfoot
      class="c-PJLV"
    >
      <tr
        class="c-hjAYDb"
      >
        <td
          class="c-cCZzXk"
        >
          Footer 1
        </td>
      </tr>
    </tfoot>
  </table>
</div>
`;
